lib/shrinkwrap.js

```diff
@@ -33,10 +36,34 @@ function shrinkwrap (args, silent, cb) {
   npm.config.set('production', true)
   readPackageTree(dir, andRecalculateMetadata(iferr(cb, function (tree) {
     var pkginfo = treeToShrinkwrap(tree, !!npm.config.get('dev') || /^dev(elopment)?$/.test(npm.config.get('also')))
-    shrinkwrap_(pkginfo, silent, cb)
+
+    readPackage(iferr(cb, function (data) {
+      chain([
+        [lifecycle, data, 'preshrinkwrap'],
```

Comment:
Given our experiences with other lifecycles, I wonder if `preshrinkwrap` shouldn't run _before_ we do the `readPackageTree` (this would mean reading the package.json prior to calling it. This would allow `preshrinkwrap` lifecycles to modify the tree, which might be desirable for some folks.

So what I'm suggesting is:

1. lifecycle: preshrinkwrap
2. readPackageTree
3. lifecycle: shrinkwrap
4. _shrinkwrap
5. lifecycle: postshrinkwrap